lib/internal/streams/from.js

```diff
@@ -75,23 +79,37 @@ function from(Readable, iterable, opts) {
   }
 
   async function next() {
-    try {
-      const { value, done } = await iterator.next();
-      if (done) {
-        readable.push(null);
-      } else {
-        const res = await value;
-        if (res === null) {
-          reading = false;
-          throw new ERR_STREAM_NULL_VALUES();
-        } else if (readable.push(res)) {
-          next();
+    for (;;) {
+      try {
+        let r;
+        if (isAysnc) {
+          r = await iterator.next();
+        } else {
+          r = iterator.next();
```

Comment:
The change here is that `iterator.next()` with a sync iterator is not `await`ed and if there are multiple `next` calls it is pushed directly without waiting for the next `next`?

Reply:
if the iterator is not async, it's not neccessary to await it